flow/CompressionUtils.cpp

```diff
@@ -81,27 +66,17 @@ StringRef CompressionUtils::compress(const CompressionFilter filter, const Strin
 	if (filter == CompressionFilter::NONE) {
 		return StringRef(arena, data);
 	}
-
-	namespace bio = boost::iostreams;
-	std::stringstream compStream;
-	std::stringstream decomStream(data.toString());
-
-	bio::filtering_streambuf<bio::input> out;
-#ifdef ZLIB_LIB_SUPPORTED
-	if (filter == CompressionFilter::GZIP) {
-		out.push(bio::gzip_compressor(bio::gzip_params(level)));
-	}
-#endif
 #ifdef ZSTD_LIB_SUPPORTED
 	if (filter == CompressionFilter::ZSTD) {
-		out.push(bio::zstd_compressor(bio::zstd_params(level)));
+		const char* src = reinterpret_cast<const char*>(data.begin());
+		size_t destSize = ZSTD_compressBound(data.size());
+		std::unique_ptr<uint8_t[]> dest = std::make_unique<uint8_t[]>(destSize);
+		size_t bytes = ZSTD_compress(dest.get(), destSize, src, data.size(), level);
+		ASSERT(!ZSTD_isError(bytes));
```

Comment:
Would it be better to `throw` an appropriate error incase underlying ZSTD call fails for some reason?